app/src/main/java/bg/zahov/app/ui/history/HistoryScreen.kt

```diff
@@ -0,0 +1,236 @@
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.CircularProgressIndicator
+import androidx.compose.material3.Icon
+import androidx.compose.material3.LocalContentColor
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.painter.Painter
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import androidx.lifecycle.viewmodel.compose.viewModel
+import bg.zahov.fitness.app.R
+import kotlin.math.min
+
+
+@Composable
+fun HistoryScreen(historyViewModel: HistoryViewModel = viewModel(), onItemClick: (String) -> Unit) {
+    val uiState by historyViewModel.uiState.collectAsStateWithLifecycle()
+    val context = LocalContext.current
+    HistoryContent(uiState.workouts, context, onItemClick = { onItemClick(it) })
+}
+
+@Composable
+fun HistoryContent(
+    workouts: List<HistoryWorkout>,
+    context: Context,
+    onItemClick: (String) -> Unit
+) {
+    AnimatedContent(
+        workouts.size,
+        transitionSpec = {
+            fadeIn(
+                animationSpec = tween(context.resources.getInteger(R.integer.animation_duration_medium))
+            ) togetherWith fadeOut(
+                targetAlpha = context.resources.getInteger(R.integer.animation_duration_medium)
+                    .toFloat()
+            )
+        },
+        label = ""
+    ) {
+        when (it) {
+            0 -> {
+                CircularProgressIndicator(
+                    modifier = Modifier,
+                    color = MaterialTheme.colorScheme.secondary,
+                    trackColor = MaterialTheme.colorScheme.surfaceVariant
+                )
+            }
+
+            else -> {
+                LazyColumn(Modifier.fillMaxSize()) {
+                    items(items = workouts, key = { it.id }) {
+                        Workout(it) {
+                            onItemClick(it)
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+@Composable
+fun Workout(item: HistoryWorkout, onItemClick: (String) -> Unit) {
+    Box(
```

Comment:
Why is this Box needed? Can we use the modifier to the column directly?

Reply:
👍